Megaparsec/Megaparsec.lean

```diff
@@ -58,21 +60,37 @@ inductive Result (S E A: Type) [Stream S] where
 | ok (x : A)
 | err (e : ParseError S E)
 
-structure Reply (S E A : Type) [Stream S] where
+structure Reply (S E A: Type) [Stream S] where
   state    : State S E
   consumed : Bool
   result   : Result S E A
 
-structure ParsecT (E S: Type) (M: Type -> Type) (A: Type) [Stream S] where
+structure ParsecT (E S: Type) [Stream S] (M: Type -> Type) [Monad M] (A: Type) where
   unParser :
     (B : Type) -> (State S E) ->
+    -- Return A with State S E and Hints into M B
     (A -> State S E -> Hints (Stream.Token S) -> M B) -> -- Consumed-OK
+    -- Report errors with State into M B
     (ParseError S E -> State S E -> M B) ->              -- Consumed-Error
+    -- Return A with State S E and Hints into M B
     (A -> State S E -> Hints (Stream.Token S) -> M B) -> -- Empty-OK
+    -- Report errors with State into M B
     (ParseError S E -> State S E -> M B) ->              -- Empty-Error
     M B
 
-def pMap (E S: Type) [Stream S] (M: Type -> Type) (U V: Type) (f: U -> V) (x: ParsecT E S M U) : ParsecT E S M V  :=
+private def run_cok (E S: Type) [Stream S] (M: Type -> Type) [Monad M] (A: Type) (a: A) (s₁: State S E) (_h: Hints (Stream.Token S)): M (Reply S E A) :=
```

Comment:
I don't like `private` defs, it feels too OOP-y. Many times in Haskell I've wanted to access a definition, but because it wasn't exported, I have no choice but to copy/paste code